### PR TITLE
#160451515 User should be able to bookmark articles

### PR DESCRIPTION
**Pull request by kampkelly**
#### What does this PR do?
This pull request adds bookmarking to articles.

#### Description of Task to be completed?
- Write tests for bookmarking articles
- Write the code to bookmark an article

#### How should this be manually tested?
After cloning the project, run `npm install`, then `npm run watch` to start the server. Using postman and as a signed in user, make requests to the following api endpoints.

To bookmark an article: /api/v1/bookmarks -> POST request (params: articleId)
To retrieve bookmarks: /api/v1/bookmarks -> GET request
To delete bookmark: /api/v1/bookmarks -> DELETE request (params: articleId)

#### Any background context you want to provide?
None
#### What are the relevant pivotal tracker stories?
#160451515
#### Questions:```
None

### CODE REVIEW

**Review by samdiano — COMMENTED** [review]
LGTM

**Review by oladimillion — COMMENTED** [review]
LGTM

**Review by onwuvic — COMMENTED** [review]
LGTM

File: server/controllers/bookmarkController.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        articleId,
+        userId: req.user.id
+      },
+      attributes: ['userId', 'articleId'],
+      include: [
+        { model: Article }
+      ]
+    })
+      .spread((bookmark, created) => {
+        if (!created) {
```
Comment (s-akinrele) [inline server/controllers/bookmarkController.js:46]: refactor this method to be more readable

**Review by s-akinrele — CHANGES_REQUESTED** [review]
I had some comments

File: server/controllers/bookmarkController.js (hunk 1 above)
Comment (kampkelly) [reply server/controllers/bookmarkController.js:46]: This has been resolved.

**Review by s-akinrele — APPROVED** [review]

**Pull request merged**